dGame/dComponents/TriggerComponent.cpp

```diff
@@ -113,22 +160,169 @@ std::vector<Entity*> TriggerComponent::GatherTargets(LUTriggers::Command* comman
+		for (auto* player : Player::GetAllPlayers()) {
+			entities.push_back(player);
+		}
+	} else if (command->target == "allNPCs") { /*UNUSED*/ }
 
 	return entities;
 }
 
-void TriggerComponent::HandleSetPhysicsVolume(Entity* targetEntity, std::vector<std::string> argArray, std::string target) {
-	PhantomPhysicsComponent* phanPhys = m_Parent->GetComponent<PhantomPhysicsComponent>();
-	if (!phanPhys) return;
+void TriggerComponent::HandleFireEvent(Entity* targetEntity, std::string args) {
+	for (CppScripts::Script* script : CppScripts::GetEntityScripts(targetEntity)) {
+		script->OnFireEventServerSide(targetEntity, m_Parent, args, 0, 0, 0);
+	}
+}
+
+void TriggerComponent::HandleDestroyObject(Entity* targetEntity, std::string args){
+	uint32_t killType;
+	GeneralUtils::TryParse<uint32_t>(args, killType);
+	targetEntity->Smash(m_Parent->GetObjectID(), static_cast<eKillType>(killType));
+}
+
+void TriggerComponent::HandleToggleTrigger(Entity* targetEntity, std::string args){
+	auto* triggerComponent = targetEntity->GetComponent<TriggerComponent>();
+	if (!triggerComponent) return;
+	triggerComponent->SetTriggerEnabled(args == "1");
+}
+
+void TriggerComponent::HandleResetRebuild(Entity* targetEntity, std::string args){
+	auto* rebuildComponent = targetEntity->GetComponent<RebuildComponent>();
+	if (!rebuildComponent) return;
+	rebuildComponent->ResetRebuild(args == "1");
+}
+
+void TriggerComponent::HandleMoveObject(Entity* targetEntity, std::vector<std::string> argArray){
+	auto position = targetEntity->GetPosition();
+
+	NiPoint3 offset = NiPoint3::ZERO;
+	GeneralUtils::TryParse<float>(argArray.at(0), offset.x);
+	GeneralUtils::TryParse<float>(argArray.at(1), offset.y);
+	GeneralUtils::TryParse<float>(argArray.at(2), offset.z);
+
+	position.x += offset.x;
+	position.y += offset.y;
+	position.z += offset.z;
+	targetEntity->SetPosition(position);
+}
 
-	phanPhys->SetPhysicsEffectActive(true);
+void TriggerComponent::HandleRotateObject(Entity* targetEntity, std::vector<std::string> argArray){
+	NiPoint3 vector = NiPoint3::ZERO;
+	GeneralUtils::TryParse<float>(argArray.at(0), vector.x);
+	GeneralUtils::TryParse<float>(argArray.at(1), vector.y);
+	GeneralUtils::TryParse<float>(argArray.at(2), vector.z);
+	NiQuaternion rotation = NiQuaternion::FromEulerAngles(vector);
+	targetEntity->SetRotation(rotation);
+}
+
+void TriggerComponent::HandlePushObject(Entity* targetEntity, std::vector<std::string> argArray){
+	auto* phantomPhysicsComponent = m_Parent->GetComponent<PhantomPhysicsComponent>();
+	if (!phantomPhysicsComponent) return;
+	phantomPhysicsComponent->SetPhysicsEffectActive(true);
+	phantomPhysicsComponent->SetEffectType(0); // push
+	phantomPhysicsComponent->SetDirectionalMultiplier(1);
+	NiPoint3 direction = NiPoint3::ZERO;
+	GeneralUtils::TryParse<float>(argArray.at(0), direction.x);
+	GeneralUtils::TryParse<float>(argArray.at(1), direction.y);
+	GeneralUtils::TryParse<float>(argArray.at(2), direction.z);
```

Comment:
A great addition to TryParse for the sake of not repeating yourself would be something like this implemented into GeneralUtils as a template specialization for TryParse
```
    template <>
	bool TryParse(const std::string& value, NiPoint3& dst) {
		return TryParse<float>(value.c_str(), dst.x) &&
				TryParse<float>(value.c_str(), dst.y) &&
				TryParse<float>(value.c_str(), dst.z);
	}
```
I will also note, the purpose of TryParse is to fail and return false if the value failed to be parsed.  It may be wise to check that you actually got the A-OK that the value was parsed correctly

Reply:
update to this - the function needs to be inlined instead and ive updated the code above to reflect that

Reply:
another update - ive realized you need to be able to pass more than one value to the function in this specialization which makes this a bit more complicated than I initially thought.  Not sure if this is exactly doable with simple changes as above since the value is a single string and not a vector.

Reply:
Looks like the best way to do this is to just overload the function so it takes 3 parameters and is not a template<>.  Makes more sense than what i suggested in the previous comment 😅